contracts/plugins/assets/angle/USDAFiatCollateral.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+pragma solidity 0.8.19;
+
+import { CollateralConfig } from "../AppreciatingFiatCollateral.sol";
+import { ERC4626FiatCollateral } from "../ERC4626FiatCollateral.sol";
+
+/**
+ * @title USDA Fiat Collateral
+ * @notice Collateral plugin for USDA (Angle)
+ * tok = stUSD (ERC4626 vault)
+ * ref = USDA
+ * tar = USD
+ * UoA = USD
+ */
+
+contract USDAFiatCollateral is ERC4626FiatCollateral {
+    /// config.erc20 must be stUSD
+    /// @param config.chainlinkFeed Feed units: {UoA/ref}
+    /// @param revenueHiding {1} A value like 1e-6 that represents the maximum refPerTok to hide
+    constructor(CollateralConfig memory config, uint192 revenueHiding)
+        ERC4626FiatCollateral(config, revenueHiding)
+    {
+        require(config.defaultThreshold != 0, "defaultThreshold zero");
+    }
+}
```

Comment:
looking at the stUSDA ERC4626 contract, and there seems to be an internal accrual happening on any deposit or withdrawal. since this accrual affects the pricePerShare() rate, we should make sure this gets updated on any call to `collateral.refresh()` (refresh() ensures that all state variables affecting price are updated so that the most accurate, up-to-date price can be calculated).

the best way to do this is probably to copy over the refresh call from the `AppreciatingFiatCollateral` and add a call to `deposit(0, address(this))` at the top of the function.

Reply:
Got it, but wondering if this is really needed for `stUSD`. View functions like `totalAssets` or `convertToAssets` already includes accruals (see https://github.com/AngleProtocol/angle-transmuter/blob/main/contracts/savings/Savings.sol#L109). So it is possible to compute up-to-date prices without having to trigger a state update, isn't it?

Reply:
yes but there are state updates performed in the collateral plugin, so we should match that with any necessary state updates in the collateral itself.